### ISSUE TITLE
Detect shared device mode

### ISSUE THREAD

**Issue by MihaMarkic** [AzureAD/microsoft-authentication-library-for-dotnet#2910]
Is there a way to detect Shared Device Mode? I need this info because when device is in this mode I should log out user after certain period of time.

**Comment by bgavrilMS** [comment]
No, we have not implemented the "Front Line Worker" scenario in Xamarin. The work needed is quite complex and we are exploring wrapping the native MSAL instead of duplicating the logic. I do not have an ETA for this.

**Comment by MihaMarkic** [comment]
Ok, thanks. But let's say if I wrap native MSAL just for this method, is it still complicated for me? Or it is just including native library and P/Invoke a single method.

**Comment by bgavrilMS** [comment]
We experimented with creating a .NET biding for MSAL for Android and I can tell you that it is very complex (many months of work). And the same for iOS seems to be even more complex.

**Comment by MihaMarkic** [comment]
> 
> 
> We experimented with creating a .NET biding for MSAL for Android and I can tell you that it is very complex (many months of work). And the same for iOS seems to be even more complex.

In general or just for using that method to check for shared device mode? I guess both, but hope only former :)

**Comment by bgavrilMS** [comment]
Doc PR to link this issue to docs - https://github.com/MicrosoftDocs/entra-docs/pull/410

**Comment by Uridel** [comment]
Im also seeing this feature being requested by our customers. We are using Maui, in the 3 years this has been requested. Has this gotten any attention, or a roadmap for when/if to expect this?

**Comment by Uridel** [comment]
Is there an update regarding this feature request? We have customers that want to use Samsung Knox and Microsoft Entra ID for front line workers. However that isn't currently possible because we cannot provide a way to detect shared device mode. Is there any indication on when this feature can be expected. As i'm quite sure the demand for this feature will increase.

**Comment by juan-ibanez-incontrol** [comment]
Any update on this issue? We have a lot of hospitals in Australia that are demanding Microsoft Entra ID in shared devices.

**Comment by Uridel** [comment]
Another year a new request for any form of update. We are getting more and more requests from hospitals to implement this.

**Comment by Thepriestdude** [comment]
Do we have any updates on this?

**Comment by noka03** [comment]
Can we bump this issue.
This is a show stopper for one of my applications using MAUI for iOS.
